pkg/bloomcompactor/tsdb.go

```diff
@@ -127,74 +135,21 @@ type forSeries interface {
 		fn func(labels.Labels, model.Fingerprint, []index.ChunkMeta),
 		matchers ...*labels.Matcher,
 	) error
-	Close() error
-}
-
-type TSDBSeriesIter struct {
-	mtx    sync.Mutex
-	f      forSeries
-	bounds v1.FingerprintBounds
-	ctx    context.Context
-
-	ch          chan *v1.Series
-	initialized bool
-	next        *v1.Series
-	err         error
-}
-
-func NewTSDBSeriesIter(ctx context.Context, f forSeries, bounds v1.FingerprintBounds) *TSDBSeriesIter {
-	return &TSDBSeriesIter{
-		f:      f,
-		bounds: bounds,
-		ctx:    ctx,
-		ch:     make(chan *v1.Series),
-	}
-}
-
-func (t *TSDBSeriesIter) Next() bool {
-	if !t.initialized {
-		t.initialized = true
-		t.background()
-	}
-
-	select {
-	case <-t.ctx.Done():
-		return false
-	case next, ok := <-t.ch:
-		t.next = next
-		return ok
-	}
-}
-
-func (t *TSDBSeriesIter) At() *v1.Series {
-	return t.next
 }
 
-func (t *TSDBSeriesIter) Err() error {
-	t.mtx.Lock()
-	defer t.mtx.Unlock()
-
-	if t.err != nil {
-		return t.err
-	}
-
-	return t.ctx.Err()
-}
-
-func (t *TSDBSeriesIter) Close() error {
-	return t.f.Close()
-}
-
-// background iterates over the tsdb file, populating the next
-// value via a channel to handle backpressure
-func (t *TSDBSeriesIter) background() {
-	go func() {
-		err := t.f.ForSeries(
-			t.ctx,
-			t.bounds,
-			0, math.MaxInt64,
-			func(_ labels.Labels, fp model.Fingerprint, chks []index.ChunkMeta) {
-
+func NewTSDBSeriesIter(ctx context.Context, f forSeries, bounds v1.FingerprintBounds) (v1.Iterator[*v1.Series], error) {
+	// TODO(salvacorts): Create a pool
+	series := make([]*v1.Series, 0, 100)
```

Comment:
it's a good idea but imo until it comes to be a problem I like to keep things simple.